code/obj/decal/cleanable.dm

```diff
@@ -737,6 +737,25 @@ var/list/blood_decal_violent_icon_states = list("floor1", "floor2", "floor3", "f
 	sample_reagent = "iron"
 	sample_verb = "scrape"
 
+/obj/decal/cleanable/rust/jen
+	icon_state = "rust_jen"
+	random_icon_states = null
+	plane = PLANE_NOSHADOW_BELOW
+	var/turf/my_turf = null
+
+	New()
+		..()
+		my_turf = get_turf_loc(src)
```

Comment:
This won't work properly if someone decides to use this decal in a non-map context because decals are pooled so `my_turf` initialization should also happen in `unpooled`. However, I don't think the `my_turf` var is necessary. Why not just use `src.loc` in attackby?

Reply:
Y'know... I hadn't thought of that, lol. Fixing!